src/main/webapp/components/navbar.vue

```diff
@@ -0,0 +1,158 @@
+<template>
+  <nav class="blue nav-bar">
+    <a href="index.html" class="bold" id="title">starfish</a>
+    <img src="../assets/starfish.png" id="logo">
+    <div id="action-items">
+      <a href="#" id="post-action">PostANote</a>
+      <div class="dropdown">
+        <button class="dropbtn" @click.stop="toggleDropdown">
+          <img id="user-profile" src="../assets/user.svg">
+        </button>
+        <div class="dropdown-content" v-if="showDropdown" v-click-outside="hideDropdown">
+          <img src="../assets/user.svg" height="115px"/>
+          <p class="bold" id="dropdown-name">{{ user.name }}</p>
+          <p id="dropdown-points">{{ user.points }} points</p>
+          <a class="dropdown-link" href="#">My Profile</a> 
+          <a class="dropdown-link" href="#">Favorite Notes</a>
+          <a class="dropdown-link" href="#">Logout</a>
+        </div>
+      </div>
+    </div> 
+  </nav>
+</template>
+
+<script>
```

Comment:
I suggest making component styles scoped so they don't interfere with other similar class names or ids
```suggestion
<script scoped>
```